### ISSUE TITLE
Prisma builds on docker image node:16 but not node:16-slim on M1 Mac

### ISSUE THREAD

**Issue by rmzg-ll** [prisma/prisma#17071]
### Bug description

Prisma fails to find a binaryTarget when built under `node:16-slim`, on an M1 Mac, but succeeds under `node:16`, both of which are based on Unbuntu Buster images.

```
Error: Unknown binaryTarget linux-arm64-openssl-undefined and no custom engine files were provided
```

I would expect it to either successfully find the right binaryTarget on both `node:16` and `node:16-slim` or fail on both.

### How to reproduce

Using a dockerfile like so:

```
########### Build Stage ############
FROM node:16-slim

# Prisma requires libssl1.1 and -slim doesn't include it, we need it at build time and run time.
RUN apt-get update && apt-get install -y libssl1.1 && rm -rf /var/lib/apt/lists/*

WORKDIR /home/apiuser

COPY ./package.json ./
COPY ./yarn.lock ./.yarnrc.yml ./
COPY ./.yarn ./.yarn
COPY ./api/package.json ./api/
COPY ./web/package.json ./web/
COPY ./packages ./packages
RUN yarn plugin import workspace-tools

# Try to avoid installing devDependencies, specifically this omits playwright and cypress which are quite large.
RUN yarn workspaces focus --production --all

COPY . ./
# Actually create the compiled api/dist files
# RUN yarn rw build api
RUN ./node_modules/.bin/rw build api
```

Run `docker build .` in a project using prisma on a M1 Mac and see the following output:

```
 => [stage-0 11/17] RUN yarn workspaces focus --production --all                                                    32.2s
 => [stage-0 12/17] COPY . ./                                                                                        0.5s
 => ERROR [stage-0 13/17] RUN ./node_modules/.bin/rw build api                                                       2.6s
------
 > [stage-0 13/17] RUN ./node_modules/.bin/rw build api:
#19 1.835 [STARTED] Generating Prisma Client...
#19 2.473 [FAILED] Command failed with exit code 1: node "/home/apiuser/node_modules/@redwoodjs/cli/node_modules/prisma/build/index.js" generate --schema="/home/apiuser/api/db/schema.prisma"
#19 2.473 [FAILED] Error: Unknown binaryTarget linux-arm64-openssl-undefined and no custom engine files were provided
#19 2.473 Command failed with exit code 1: node "/home/apiuser/node_modules/@redwoodjs/cli/node_modules/prisma/build/index.js" generate --schema="/home/apiuser/api/db/schema.prisma"
#19 2.473 Error: Unknown binaryTarget linux-arm64-openssl-undefined and no custom engine files were provided
------
executor failed running [/bin/sh -c ./node_modules/.bin/rw build api]: exit code: 1
```

If you change the line `FROM node:16-slim` to `FROM node:16` and nothing else, the `docker build` command completes successfully. Which is weird.

### Expected behavior

I would expect it to either successfully find the right binaryTarget on both `node:16` and `node:16-slim` or fail on both.

### Prisma information

None

### Environment & setup

Environment:
Docker image node:16 and node:16-slim

### Prisma Version

package.json:

```
"@prisma/client": "4.3.1",
"prisma": "4.3.1",
```

**Comment by annibuliful** [comment]
Because `node:16-slim` is not be installed openSSL which is prisma engine need it as a dependency so you need to use `node:16` instead

**Comment by rmzg-ll** [comment]
> Because `node:16-slim` is not be installed openSSL which is prisma engine need it as a dependency so you need to use `node:16` instead

The second line in the Dockerfile is `RUN apt-get update && apt-get install -y libssl1.1 && rm -rf /var/lib/apt/lists/*`, are you saying this isn't sufficient?

**Comment by annibuliful** [comment]
I tried to use the same way but it does not solve

**Comment by Jolg42** [comment]
@annibuliful @rmzg-ll Did you try using Prisma `4.8.0`? We changed the related logic recently.
It would be interesting to know how it fails (differently or not) in `4.8.0`

**Comment by jkomyno** [comment]
Here's some additional context about this issue.

The `node:slim` Docker image (and variants, like `node:lts-slim`, `node:16-slim`, etc) is a multiplatform image.
This means that on `arm64` machines, like macOS with silicon (M1, M2) CPUs, you're actually using `arm64v8/node:slim` (and variants, like `arm64v8/node:lts-slim`). 

Although we build and support the Prisma engines for `libc`-based Linux distros (i.e., all main distros except `alpine`) on the `arm64` architecture, we had a bug that prevented us from understanding the version of `openssl` installed in the system (which is required by Prisma's query-engine and the Prisma Client).

**Comment by jkomyno** [comment]
Hi @rmzg-ll, may you please tell us whether the provided solution works for you?
The `larger, stable Prisma release` mentioned [here](https://github.com/prisma/prisma/issues/17071#issuecomment-1370042999) is now available, please use `prisma@4.9.0`

**Comment by jkomyno** [comment]
Closing this as a solution is provided in [this comment](https://github.com/prisma/prisma/issues/17071#issuecomment-1370051143), and Prisma will also start officially supporting Linux Alpine on `arm64` architectures starting from `prisma@4.10.0`, which will be released in a few hours.
